pkg/cmd/get/get_preview_test.go

```diff
@@ -0,0 +1,193 @@
+package get_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/jenkins-x/jx/pkg/kube"
+
+	"github.com/jenkins-x/jx/pkg/cmd/get"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/testhelpers"
+	"github.com/jenkins-x/jx/pkg/gits"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	jenkinsv1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	clientmocks "github.com/jenkins-x/jx/pkg/cmd/clients/mocks"
+	helm_test "github.com/jenkins-x/jx/pkg/helm/mocks"
+	resources_test "github.com/jenkins-x/jx/pkg/kube/resources/mocks"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCurrentPreviewUrlWithoutEnvVars(t *testing.T) {
+	ro := os.Getenv("REPO_OWNER")
```

Comment:
There seems to be a lot of duplidated setup code in each test. I'd extract a setup and teardown function which move this out of the test function. teardown can be called in the defer clause. 

This helps draw attention to the actual test.

Reply:
Personally, I'd write Ginkgo test, since imo it more expressive, but that might be a matter of case.